ci/audit-logs-s3.py

```
@@ -0,0 +1,126 @@
+        for page in range(total_pages):
+            result= subprocess.check_output(
+            "cf curl '/v3/audit_events?created_ats[gt]=" + str(start) + "&created_ats[lt]=" + str(end) + "&order_by=created_at&page=" + str(page+1)+"'",
+            universal_newlines=True,
+            shell=True,
+            )
+            data= json.loads(result)
+            result_data=data.get("resources",{})
+            audit_logs.extend(result_data)
+        return audit_logs
+    else:
+        return "empty"
+
+@functools.cache
+def get_cf_entity_name(entity, guid):
+    """
+    Retrieves the name of a CF entity from a GUID.
+    """
+    if not guid:
+        return
+    elif 'guid' in guid:
+        guid = ast.literal_eval(guid)['guid']
+        cf_json = subprocess.check_output(
+            "cf curl /v3/" + entity + "/" + guid,
+            universal_newlines=True,
+            shell=True,
+        )
+        cf_data = json.loads(cf_json)
+        return cf_data.get("name", "N/A")
+    else:
+        return "None"
+
+
+def upload_to_s3(bucket_name, object_name, data):
+    try:
+        body = '\n'.join([
+                    json.dumps({
+                           **{k: v for k,v in item.items() if k not in ["links"]},
+                           "organization_name": get_cf_entity_name("organizations", f"{item['organization']}"),
+                           "space_name": get_cf_entity_name("spaces", f"{item['space']}")
+                           })
+                          for item in data
+                          ])
+        s3_client.put_object(
+            Bucket=bucket_name,
+            Key=object_name,
+            Body=body,
+            ContentType='application/json'
+            )
+        print("success for time "+ str(start_time))
+    except Exception as e:
+        print(traceback.format_exc())
+        print(f'Error upload file to S3 for time starting' + str(start_time) + " and end time" + str(end_time))
+        exit(1)
```

```suggestion
        print(f'Error upload file to S3 for time starting' + str(start_time) + " and end time" + str(end_time))
        raise e
```

I think as long as we re-raise, `main()` should throw the exception and exit 1